GVFS/GVFS.Upgrader/GVFS.Upgrader.csproj

```diff
@@ -1,137 +1,56 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
-  <Import Project="$(SolutionDir)\GVFS\GVFS.Build\GVFS.cs.props" />
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="..\GVFS.Build\GVFS.cs.props" />
+
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{AECEC217-2499-403D-B0BB-2962B9BE5970}</ProjectGuid>
     <OutputType>Exe</OutputType>
-    <RootNamespace>GVFS.Upgrader</RootNamespace>
     <AssemblyName>GVFS.Upgrader</AssemblyName>
-    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
-    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
-    <NuGetPackageImportStamp>
-    </NuGetPackageImportStamp>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x64'">
-    <DebugSymbols>true</DebugSymbols>
-    <DefineConstants>DEBUG;TRACE</DefineConstants>
-    <DebugType>full</DebugType>
+    <TargetFrameworks>net461;netcoreapp2.1</TargetFrameworks>
+
+    <Platforms>x64</Platforms>
     <PlatformTarget>x64</PlatformTarget>
-    <ErrorReport>prompt</ErrorReport>
-    <Prefer32Bit>false</Prefer32Bit>
+    <RootNamespace>GVFS.Upgrader</RootNamespace>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>
+    <RuntimeIdentifiers>osx-x64</RuntimeIdentifiers>

```

Comment:
This `RuntimeIdentifier` probably does not belong in this common PropertyGroup that is included for both `net461` and `netcoreapp2.1`.

Reply:
You can set the identifier conditionally on the TF.

```xml
<TargetFrameworks>netcoreapp2.1;net461</TargetFrameworks>
<RuntimeIdentifiers>win-x64;osx-x64;linux-x64</RuntimeIdentifiers>
<RuntimeIdentifier Condition="'$(TargetFramework)'=='net461'">win-x64</RuntimeIdentifier>
```

Here when building the TF as `net461` you're also forcing the RID to be `win-x64`.
For TF `netcoreapp2.1` you'll need to specify which OS you're targeting still.

Reply:
What @mjcheetham proposed seems like it should work, but I bet VSMac won't be able to parse it. 

It sounds like most of us are moving towards using VSCode going forward, so maybe it's alright to take this proposed change if VSMac can't handle this MSBuild structure.

Reply:
> It sounds like most of us are moving towards using VSCode going forward, so maybe it's alright to take this proposed change if VSMac can't handle this MSBuild structure.

Please don't break VSMac until we've decided that it's the preferred editor for Mac development and have documented how to configure VSCode to build/debug/etc. VFS4G.

Reply:
> Please don't break VSMac

This was my goal. If I couldn't get this to work with a single csproj file, I was going to fall back to having 2 separate csproj files.

Reply:
> Please don't break VSMac

This was my goal. If I couldn't get this to work with a single csproj file, I was going to fall back to having 2 separate csproj files.